Core/src/org/sleuthkit/autopsy/corecomponents/AutopsyOptionsPanel.java

```diff
@@ -476,6 +485,24 @@ private boolean isMemFieldValid() {
         return true;
     }
 
+    /**
+     * Check if the logFileCount field is valid.
+     * 
+     * @return true if the logFileCount is valid false if it is not
+     */
+    private boolean isLogNumFieldValid() {
+        String count = logFileCount.getText();
+        logNumAlert.setText("");
+        if (count.length() == 0) {
+            logNumAlert.setText(Bundle.AutopsyOptionsPanel_logNumAlert_noValueEntered_text());
+            return false;
+        } 
+        if (count.replaceAll("[^\\d]", "").length() != count.length() || count.replaceFirst("^0+", "").length() == 0) {
+            logNumAlert.setText(Bundle.AutopsyOptionsPanel_logNumAlert_invalidInput_text());
+            return false;
+        }
+        return true;      
```

Comment:
@rcordovano  Here the validation code is similar to isMemFieldValid(). The user interface looks better, because this way the Java NumberFormatException window is not pop up, only show the warning message. But this is not checking if the input is larger than the Integer.MAX_VALUE. 
I like the following way to check if a number is a positive integer, it will include checking empty input, fraction, or number>Integer.MAX_VALUE. 

        String count = logFileCount.getText();
         try {
             int count_num = Integer.parseInt(count);
             if (count_num < 1) {
                 logNumAlert.setText(Bundle.AutopsyOptionsPanel_logFileCountAlert_text());
                 return false;
             } 
         } catch (NumberFormatException e) {
             logNumAlert.setText(Bundle.AutopsyOptionsPanel_logFileCountAlert_text());
             return false;
         }
But this will popup Java Exception window besides show the warning message if the input is invalid. Please let me know which way is better here.

Reply:
We definitely do not want a pop up as well as a message, but we should still try parsing the int and catching the exception. I am not seeing how the int parsing code can be throwing an exception we can't catch , so please put the code back to this way and show me how the pop up is happening, thanks.